internal/pkg/aws/session/session.go

```diff
@@ -30,11 +30,23 @@ func userAgentHandler() request.NamedHandler {
 	}
 }
 
-// Factory holds methods to create sessions.
-type Factory struct{}
+// Provider provides methods to create sessions.
+// Once a session is created, it's cached locally so that the same session is not re-created.
+type Provider struct {
+	defaultSess *session.Session
+}
+
+// NewProvider initializes a new session Provider with empty caches.
```

Comment:
Since the caches already been removed, maybe this comment should be updated too.

Reply:
wamp, I merged lol but I'll fix this in the next PR!